src/app/auth/views/SignUpView/SignUpView.tsx

```diff
@@ -307,131 +325,181 @@ const SignUpView = (props: SignUpViewProps): JSX.Element => {
-            required={true}
-            minLength={{ value: 1, message: 'Name must not be empty' }}
-            error={errors.name}
-          />
-
-          <BaseInput
-            className="mb-2.5"
-            placeholder="Lastname"
-            label="lastname"
-            type="text"
-            icon={<UilUser className="w-4" />}
-            register={register}
-            required={true}
-            minLength={{ value: 1, message: 'Lastname must not be empty' }}
-            error={errors.lastname}
-          />
-
-          <BaseInput
-            className="mb-2.5"
-            placeholder="Email"
-            label="email"
-            type="email"
-            disabled={hasEmailParam}
-            icon={<UilEnvelope className="w-4" />}
-            register={register}
-            required={true}
-            minLength={{ value: 1, message: 'Email must not be empty' }}
-            pattern={{ value: emailRegexPattern, message: 'Email not valid' }}
-            error={errors.email}
-          />
-
-          <BaseInput
-            className="mb-2.5"
-            placeholder="Password"
-            label="password"
-            type={showPassword ? 'text' : 'password'}
-            icon={
-              password ? (
-                showPassword ? (
-                  <UilEyeSlash className="w-4" onClick={() => setShowPassword(false)} />
-                ) : (
-                  <UilEye className="w-4" onClick={() => setShowPassword(true)} />
-                )
-              ) : (
-                <UilLock className="w-4" />
-              )
-            }
-            register={register}
-            required={true}
-            minLength={{ value: 8, message: 'The password must be at least 8 characters long' }}
-            error={errors.password}
-            pattern={{
-              value: isValidPasswordRegex,
-              message: 'The password must contain lowercase/uppercase letters and at least a number',
-            }}
-          />
-
-          <BaseInput
-            className="mb-2.5"
-            placeholder="Confirm new password"
-            label="confirmPassword"
-            type={showConfirmPassword ? 'text' : 'password'}
-            icon={
-              confirmPassword ? (
-                showConfirmPassword ? (
-                  <UilEyeSlash className="w-4" onClick={() => setShowConfirmPassword(false)} />
-                ) : (
-                  <UilEye className="w-4" onClick={() => setShowConfirmPassword(true)} />
-                )
-              ) : (
-                <UilLock className="w-4" />
-              )
-            }
-            register={register}
-            required={true}
-            minLength={{ value: 1, message: 'Password must not be empty' }}
-            error={errors.confirmPassword}
-          />
-
-          <div className="mt-1 mb-2">
-            <span className="text-red-60 text-sm font-medium">{bottomInfoError}</span>
+    <>
+      {isDoingRegister && (
```

Comment:
Try to use a ternary operator here like: `isDoingRegister? <Component1>: <Component2>` in order to keep related components together. This change also prevents React of doing both checks to determine if the component should be rendered when one check is enough